types/jest/index.d.ts

```diff
@@ -543,7 +543,7 @@ declare namespace jest {
          * This ensures that a value matches the most recent snapshot.
          * Check out [the Snapshot Testing guide](http://facebook.github.io/jest/docs/snapshot-testing.html) for more information.
          */
-        toMatchSnapshot(snapshotName?: string): R;
+        toMatchSnapshot(propertyMatchers?: {}, snapshotName?: string): R;
```

Comment:
Both API's are[ supported by Jest 23](https://github.com/facebook/jest/blob/master/packages/jest-snapshot/src/index.js#L57). if we do what you submitted we would introduce breaking change to Typescript users.

Instead we should new API via overloads

```ts
toMatchSnapshot(propertyMatchers?: object, snapshotName?: string): R;
/** @deprecated **/
toMatchSnapshot(snapshotName?: string): R;
```

Reply:
Also we can be much smarter here and introduce stricter props checking:

```ts
/**
         * This ensures that a value matches the most recent snapshot.
         * Check out [the Snapshot Testing guide](http://facebook.github.io/jest/docs/snapshot-testing.html) for more information.
         */
        toMatchSnapshot<T extends {[P in keyof R]: Expect['any']}>(propertyMatchers: Partial<T>, snapshotName?: string): R;
        /**
         * @deprecated
         */
        toMatchSnapshot(snapshotName?: string): R;
```

which needs update of `Expect` interface

```ts
interface Expect {
  /**
   * The `expect` function is used every time you want to test a value.
   * You will rarely call `expect` by itself.
   *
   * @param actual The value to apply matchers against.
   */
  <T = any>(actual: T): Matchers<T>;
}
```

Also please add tests for this

```ts
expect({}).toMatchSnapshot();
expect({}).toMatchSnapshot('snapshotName');
expect({
  one: 1,
  two: '2',
  date: new Date(),
}).toMatchSnapshot({ one: expect.any(Number), date: expect.any(Date) });
```

![jest-typings-improved-property-matchers](https://user-images.githubusercontent.com/1223799/41810799-5c82d55e-7704-11e8-86d4-c2b26808bf92.gif)